homeassistant/components/squeezebox/coordinator.py

```diff
@@ -0,0 +1,106 @@
+"""DataUpdateCoordinator for the Squeezebox integration."""
+
+from asyncio import timeout
+from datetime import timedelta
+import logging
+import re
+
+from pysqueezebox import Server
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+from homeassistant.util import dt as dt_util
+
+from .const import (
+    SENSOR_UPDATE_INTERVAL,
+    STATUS_API_TIMEOUT,
+    STATUS_SENSOR_LASTSCAN,
+    STATUS_SENSOR_NEEDSRESTART,
+    STATUS_SENSOR_NEWPLUGINS,
+    STATUS_SENSOR_NEWVERSION,
+    STATUS_SENSOR_RESCAN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+# Sample status
+# {'info total genres': 134, 'newversion': 'A new version of Logitech Media Server is available (8.5.2 - 0). <a href="updateinfo.html?installerFile=/var/lib/squeezeboxserver/cache/updates/logitechmediaserver_8.5.2_amd64.deb" target="update">Click here for further information</a>.', 'httpport': '9000', 'lastscan': '1718431678', 'info total artists': 1954, 'info total albums': 1699, 'player count': 3, 'uuid': 'blargs-blarg-blargle', 'other player count': 2, 'version': '8.5.1', 'ip': '192.168.78.20', 'info total duration': 5036321.37400003, 'mac': '44:1e:a1:3b:78:23', 'info total songs': 20479}
+# {'lastscan': '1720418978', 'version': '8.5.1', 'newversion': 'A new version of Logitech Media Server is available (8.5.2 - 0). <a href="updateinfo.html?installerFile=/var/lib/squeezeboxserver/cache/updates/logitechmediaserver_8.5.2_amd64.deb" target="update">Click here for further information</a>.', 'uuid': 'wobble', 'mac': '44:1e:a1:3b:78:23', 'ip': '192.168.78.20', 'httpport': '9000', 'info total albums': 1702, 'info total artists': 1955, 'info total genres': 135, 'info total songs': 20575, 'info total duration': 5057231.58200003, 'player count': 4, 'other player count': 2}
+# {'ip': '192.168.78.86', 'info total duration': 364692.599999999, 'lastscan': '1720450648', 'other player count': 4, 'info total genres': 25, 'newplugins': 'Plugins have been updated - Restart Required (Material Skin)', 'uuid': 'wibble', 'info total albums': 69, 'player count': 2, 'needsrestart': 1, 'version': '8.5.2', 'info total songs': 1508, 'info total artists': 117, 'httpport': '9000'}
```

Comment:
Make unit tests, put this in there. Profit.

Reply:
Unresolved

Reply:
Comments are gone and I dont have the understanding yet to do unit tests and will run  of of time very shortlly